angular-ngrx-scss/src/app/shared/components/repo-controls/repo-filter-dropdown/repo-filter-dropdown.component.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  Output,
+} from '@angular/core';
+import { trigger, transition, style, animate } from '@angular/animations';
+
+export interface FilterOption {
+  label: string;
+  value: string;
+}
+
+@Component({
+  selector: 'app-repos-filter-dropdown',
+  templateUrl: './repo-filter-dropdown.component.html',
+  styleUrls: ['./repo-filter-dropdown.component.scss'],
+  animations: [
+    trigger('openClose', [
+      transition(':enter', [
+        style({ opacity: 0, transform: 'scale(0.95)' }),
+        animate('100ms ease-out', style({ opacity: 1, transform: 'scale(1)' })),
+      ]),
+      transition(':leave', [
+        animate('75ms', style({ opacity: 0, transform: 'scale(0.95)' })),
+      ]),
+    ]),
+  ],
+})
+export class RepoFilterDropdownComponent {
+  @Input() name = '';
+  @Input() description = '';
+  @Input() current?: string | null = '';
+  @Input() items: FilterOption[] = [];
+  @Input() toggle? = false;
+
+  @Output() setFilter: EventEmitter<string> = new EventEmitter();
+
+  isOpen = false;
+
+  constructor(private elRef: ElementRef) {}
+
+  toggleMenu() {
+    this.isOpen = !this.isOpen;
+  }
+
+  handleSetFilterClick(label: string) {
+    if (label === this.current) {
+      if (this.toggle) {
+        this.setFilter.emit('');
+      } else {
+        this.setFilter.emit(this.items[0].value);
+      }
+    } else {
+      this.setFilter.emit(label);
+    }
+    this.isOpen = false;
+  }
+
+  handleClearFilterClick() {
+    this.setFilter.emit(this.items[0].value);
+    this.isOpen = false;
+  }
+
+  // TODO: maybe convert to directive
+  @HostListener('document:click', ['$event'])
+  onClick(event: PointerEvent) {
+    if (!this.elRef.nativeElement.contains(event.target)) {
+      this.isOpen = false;
+    }
+  }
```

Comment:
suggestion: This should be a directive as we can use this functionality elsewhere in a re-usable fashion. I think this ticket would be fine for introducing this behavior. Specifically the navigation bar dropdown could use it.

Reply:
I would agree that we could make the directive, but think I'd also like that in another ticket. Then when it's made, it could also update this and the nav bar to use it. 

- [ ] Please make the ticket to create this directive so we don't forget

Reply:
New issue created.

https://github.com/thisdot/starter.dev-github-showcases/issues/492